include/systems/eigen_system.h

```diff
@@ -243,7 +243,8 @@ class EigenSystem : public System
   inline const SparseMatrix<Number> & get_precond_matrix() const
     {
       libmesh_assert(precond_matrix);
-      libmesh_assert(!_use_shell_matrices);
+      libmesh_assert(_use_shell_matrices);
```

Comment:
Is there never cause to use a non-shell matrix but also use a (perhaps sparser) preconditioner matrix?

Reply:
@fdkong what are your thoughts on this? I made this change after a convo with you.

Reply:
Ping @fdkong

Reply:
Talked with Fande today - these assertions are valid